tools/jvm/src/main/scala/org/scalajs/core/tools/optimizer/ScalaJSClosureOptimizer.scala

```diff
@@ -151,34 +167,223 @@ object ScalaJSClosureOptimizer {
+    def this(
+        output: WritableVirtualJSFile,
+        cache: Option[WritableVirtualTextFile] = None,
+        bypassLinkingErrors: Boolean = false,
+        checkIR: Boolean = false,
+        unCache: Boolean = true,
+        disableOptimizer: Boolean = false,
+        batchMode: Boolean = false,
+        wantSourceMap: Boolean = false,
+        prettyPrint: Boolean = false,
+        relativizeSourceMapBase: Option[URI] = None,
+        noWarnMissing: Seq[ScalaJSOptimizer.NoWarnMissing] = Nil) = {
+
+      this(
+        output = output,
+        cache = cache,
+        bypassLinkingErrors = bypassLinkingErrors,
+        checkIR = checkIR,
+        unCache = unCache,
+        disableOptimizer = disableOptimizer,
+        batchMode = batchMode,
+        wantSourceMap = wantSourceMap,
+        prettyPrint = prettyPrint,
+        relativizeSourceMapBase = relativizeSourceMapBase,
+        noWarnMissing = noWarnMissing,
+        customOutputWrapper = ("", ""))
+    }
+
+    def withCache(cache: Option[WritableVirtualTextFile]): Config =
+      copyWith(cache = cache)
+
+    def withBypassLinkingErrors(bypassLinkingErrors: Boolean): Config =
+      copyWith(bypassLinkingErrors = bypassLinkingErrors)
+
+    def withCheckIR(checkIR: Boolean): Config =
+      copyWith(checkIR = checkIR)
+
+    def withUnCache(unCache: Boolean): Config =
+      copyWith(unCache = unCache)
+
+    def withDisableOptimizer(disableOptimizer: Boolean): Config =
+      copyWith(disableOptimizer = disableOptimizer)
+
+    def withBatchMode(batchMode: Boolean): Config =
+      copyWith(batchMode = batchMode)
+
+    def withWantSourceMap(wantSourceMap: Boolean): Config =
+      copyWith(wantSourceMap = wantSourceMap)
+
+    def withPrettyPrint(prettyPrint: Boolean): Config =
+      copyWith(prettyPrint = prettyPrint)
+
+    def withRelativizeSourceMapBase(relativizeSourceMapBase: Option[URI]): Config =
+      copyWith(relativizeSourceMapBase = relativizeSourceMapBase)
+
+    def withNoWarnMissing(noWarnMissing: Seq[ScalaJSOptimizer.NoWarnMissing]): Config =
+      copyWith(noWarnMissing = noWarnMissing)
+
+    def withCustomOutputWrapper(customOutputWrapper: (String, String)): Config =
+      copyWith(customOutputWrapper = customOutputWrapper)
+
+    @deprecated("Not a case class anymore", "0.6.5")
+    def copy(
+        output: WritableVirtualJSFile = this.output,
+        cache: Option[WritableVirtualTextFile] = this.cache,
+        bypassLinkingErrors: Boolean = this.bypassLinkingErrors,
+        checkIR: Boolean = this.checkIR,
+        unCache: Boolean = this.unCache,
+        disableOptimizer: Boolean = this.disableOptimizer,
+        batchMode: Boolean = this.batchMode,
+        wantSourceMap: Boolean = this.wantSourceMap,
+        prettyPrint: Boolean = this.prettyPrint,
+        relativizeSourceMapBase: Option[URI] = this.relativizeSourceMapBase,
+        noWarnMissing: Seq[ScalaJSOptimizer.NoWarnMissing] = this.noWarnMissing
+    ): Config = {
```

Comment:
Leave `): Config = {` at the end of the previous line, even it goes over 80 chars, as recommended in https://github.com/scala-js/scala-js/blob/master/CODINGSTYLE.md#method-definition